diagnostics/model_accuracy/verify_inference_model.py

```diff
@@ -84,28 +61,22 @@ def run_inference(model_name, models_dir):
                                  '/image_recognition.patch')
             p = command_executor(data[i]["cmd"])
             os.chdir(pwd)
-            return model_name, p
+            return model_name, p   
 
-
-def check_accuracy(model, p, tolerance=0.001):
+def check_accuracy(model, p,json_file_name, tolerance=0.001):
     #check if the accuracy of the model inference matches with the published numbers
     #Accuracy values picked up from here https://github.com/tensorflow/models/tree/master/research/slim
-    accuracy = \
-    '[{"model_name" : "Inception_v4", "accuracy" : "0.802"},\
-     {"model_name" : "ResNet50_v1", "accuracy" : "0.752"}, \
-     {"model_name" : "MobileNet_v1", "accuracy" : "0.709"}]'
-
-    data = json.loads(accuracy)
 
+    data = parse_json(args.json_file_name)
     for line in p.splitlines():
         print(line.decode())
         if ('eval/Accuracy'.encode() in line):
             accuracy = re.split("eval/Accuracy", line.decode())[1]
-            top1_accuracy = re.search(r'\[(.*)\]', accuracy).group(1)
+            top1_accuracy = re.search(r'\[([0-9.]+)]', accuracy).group(1)
         #for now we just validate top 1 accuracy, but calculating top5 anyway.
         if ('eval/Recall_5'.encode() in line):
             accuracy = re.split("eval/Recall_5", line.decode())[1]
-            top5_accuracy = float(re.search("\[(.*?)\]", accuracy).group(1))
+            top5_accuracy = re.search(r'\[([0-9.]+)]', accuracy).group(1)
```

Comment:
See above comment for how to better rewrite the re.search and drop the re.split

Reply:
@crlishka thanks Chris.
Changed to:
is_match = re.search(r'.+eval/Recall_5[([0-9.]+)]', line.decode())
if is_match and len(is_match.groups()) > 0 :
top5_accuracy = is_match.group(1)